### PR TITLE
Alias all functions ending in `?`

### PR DESCRIPTION
**Pull request by xiongtx**
Address #100.

### CODE REVIEW

File: s.el [hunk 1]
```diff
@@ -627,5 +610,28 @@ is appended, otherwise PREFIX is used as both prefix and
...
+(defalias 's-matches-p 's-matches?)
+(defalias 's-mixedcase-p 's-mixedcase?)
+(defalias 's-numeric-p 's-numeric?)
+(defalias 's-prefix-p 's-starts-with?)
+(defalias 's-prefix? 's-starts-with?)
+(defalias 's-present-p 's-present?)
+(defalias 's-starts-with-p 's-starts-with?)
+(defalias 's-suffix-p 's-ends-with?)
+(defalias 's-suffix? 's-ends-with?)
+(defalias 's-uppercase-p 's-uppercase?)
```
Comment (Wilfred) [inline s.el]: Thanks for doing this :)

Does it make sense to move all the aliases to a separate part of the file? It seems clearer to me to keep the logical functionality together.

Comment (xiongtx) [inline s.el]: I think it's helpful to group all aliases together so we can easily see what's been aliased and what hasn't. It was slightly annoying going through the file to figure that out the first time.

Comment (Wilfred) [inline s.el]: Alternatively, I suppose we could define a `s-defun` macro that automatically sets up these aliases. What do you think?

Comment (xiongtx) [inline s.el]: *Shrug* It's unidiomatic Emacs Lisp to have functions end in `?` anyway; I chalk it up to the author's Clojure background, but it's not something we should encourage.

**Comment by xiongtx** [comment]
Ping @magnars

**Comment by magnars** [comment]
Thanks for the pull request - and the ping.

**Pull request merged**